src/operations/contract.lisp

```diff
@@ -263,12 +262,20 @@ If we have a non-null peduncle edge (F -> C above), then we need to tell its sou
   "Tell the blossom itself what's up."
   (let ((frame (peek (process-data-stack supervisor))))
     (with-slots (fresh-blossom peduncle-edge path petal-child-edges) frame
-      (sync-rpc (make-message-set-up-blossom
-                 :peduncle-edge peduncle-edge
-                 :petals path
-                 :petal-children petal-child-edges)
-          (set-up-result fresh-blossom)
-        nil))))
+      ;; see `COMPUTE-BLOSSOM-PATHS' above to understand why the source node of
+      ;; the first edge in the `path' corresponds to the node closest to the root
+      (let ((rootmost-cycle-node (blossom-edge-source-node (first path))))
+        ;; determine the `DRYAD' of that "rootmost" node in the cycle
+        (sync-rpc (make-message-values :values '(dryad))
+            ((dryad-address) rootmost-cycle-node)
+          (setf (supervisor-node-dryad supervisor) dryad-address)
+          (sync-rpc (make-message-set-up-blossom
+                     :peduncle-edge peduncle-edge
+                     :petals path
+                     :petal-children petal-child-edges
+                     :dryad dryad-address)
+              (set-up-result fresh-blossom)
+            nil))))))
```

Comment:
I know this isn't yours, and my Lisp opinions are a little rusty, but I think `(values)` or even `(finish-with-scheduling)` might be more idiomatic here. Just sayin'.